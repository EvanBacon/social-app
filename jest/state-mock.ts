import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
