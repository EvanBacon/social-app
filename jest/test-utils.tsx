import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
