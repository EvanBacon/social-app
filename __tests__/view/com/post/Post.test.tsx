import React from 'react'
import {fireEvent, render} from '../../../../jest/test-utils'
import {PostThreadViewModel} from '../../../../src/state/models/post-thread-view'
import {Post} from '../../../../src/view/com/post/Post'
import {
  mockedPostThreadViewStore,
  mockedShellStore,
} from '../../../../__mocks__/state-mock'

describe('Post', () => {
  const mockedProps = {
    uri: 'testuri',
    initView: mockedPostThreadViewStore,
    showReplyLine: false,
  }
  afterAll(() => {
    jest.clearAllMocks()
  })

  it('renders ActivityIndicator when loading', async () => {
    const {findByTestId} = render(
      <Post
        {...{
          ...mockedProps,
          initView: {
            ...mockedPostThreadViewStore,
            isLoading: true,
          } as PostThreadViewModel,
        }}
      />,
    )

    const postLoading = await findByTestId('postLoading')
    expect(postLoading).toBeTruthy()
  })

  it('renders ErrorMessage on error', async () => {
    const {findByTestId} = render(
      <Post
        {...{
          ...mockedProps,
          initView: {
            ...mockedPostThreadViewStore,
            hasError: true,
          } as PostThreadViewModel,
        }}
      />,
    )

    const postError = await findByTestId('postError')
    expect(postError).toBeTruthy()
  })

  it('renders and clicks reply button', async () => {
    const {findByTestId} = render(<Post {...mockedProps} />)
    const postCtrlsReplyButton = await findByTestId('postCtrlsReplyButton')
    expect(postCtrlsReplyButton).toBeTruthy()

    fireEvent.press(postCtrlsReplyButton)
    expect(mockedShellStore.openComposer).toHaveBeenCalled()
  })

  it('renders and presses repost button', async () => {
    const {findByTestId} = render(<Post {...mockedProps} />)

    const postCtrlsToggleRepostButton = await findByTestId(
      'postCtrlsToggleRepostButton',
    )
    expect(postCtrlsToggleRepostButton).toBeTruthy()

    fireEvent.press(postCtrlsToggleRepostButton)
    expect(mockedPostThreadViewStore.thread!.toggleRepost).toHaveBeenCalled()
  })

  it('renders and presses upvote button', async () => {
    const {findByTestId} = render(<Post {...mockedProps} />)

    const postCtrlsToggleUpvoteButton = await findByTestId(
      'postCtrlsToggleUpvoteButton',
    )
    expect(postCtrlsToggleUpvoteButton).toBeTruthy()

    fireEvent.press(postCtrlsToggleUpvoteButton)
    expect(mockedPostThreadViewStore.thread!.toggleUpvote).toHaveBeenCalled()
  })

  it('renders muted view', async () => {
    const {findByTestId} = render(<Post {...mockedProps} />)

    const mutedPostView = await findByTestId('mutedPostView')
    expect(mutedPostView).toBeTruthy()
  })

  it('renders postShowReplyLine view', async () => {
    const {findByTestId} = render(
      <Post {...{...mockedProps, showReplyLine: true}} />,
    )

    const postShowReplyLineView = await findByTestId('postShowReplyLineView')
    expect(postShowReplyLineView).toBeTruthy()
  })

  it('matches snapshot', () => {
    const page = render(<Post {...mockedProps} />)
    expect(page).toMatchSnapshot()
  })
})
