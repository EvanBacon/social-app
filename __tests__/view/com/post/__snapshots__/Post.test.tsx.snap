// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderTopWidth": 1,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "borderColor": "#e4e2e2",
        },
        undefined,
      ]
    }
    testID="linkButton-/profile/test.handle/post/"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "width": 60,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="linkButton-/profile/test.handle"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={52}
            bbWidth={52}
            focusable={false}
            height={52}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 52,
                  "width": 52,
                },
              ]
            }
            vbHeight={100}
            vbWidth={100}
            width={52}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -2424577,
                      1,
                      -6862849,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="grad"
                  x1="0"
                  x2="1"
                  y1="0"
                  y2="1"
                />
              </RNSVGDefs>
              <RNSVGCircle
                cx="50"
                cy="50"
                fill={
                  Array [
                    1,
                    "grad",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="50"
              />
              <RNSVGText
                content={null}
                dx={Array []}
                dy={Array []}
                fill={4294967295}
                font={
                  Object {
                    "fontSize": "50",
                    "fontWeight": "bold",
                    "textAnchor": "middle",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "50",
                  ]
                }
                y={
                  Array [
                    "67",
                  ]
                }
              >
                <RNSVGTSpan
                  content="T"
                  dx={Array []}
                  dy={Array []}
                  font={Object {}}
                  rotate={Array []}
                  x={Array []}
                  y={Array []}
                />
              </RNSVGText>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "flexDirection": "row",
              "paddingBottom": 2,
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "maxWidth": "80%",
                "opacity": 1,
                "paddingRight": 5,
              }
            }
            testID="linkButton-/profile/test.handle"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                    "letterSpacing": 0.15,
                  },
                  Array [
                    Object {
                      "color": "#000000",
                    },
                  ],
                ]
              }
            >
              test.handle
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontSize": 15,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                    },
                    Array [
                      Object {
                        "color": "#645454",
                      },
                    ],
                  ]
                }
              >
                 
                test.handle
              </Text>
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 15,
                  "fontWeight": "400",
                  "letterSpacing": 0.15,
                },
                Array [
                  Object {
                    "paddingRight": 5,
                  },
                  Object {
                    "color": "#645454",
                  },
                ],
              ]
            }
          >
            · 
            Invalid Date
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "marginBottom": 2,
              },
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <
            icon="reply"
            size={9}
            style={
              Array [
                Object {
                  "color": "#645454",
                },
                Object {
                  "marginRight": 5,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                },
                Array [
                  Object {
                    "color": "#645454",
                  },
                  Object {
                    "marginRight": 2,
                  },
                ],
              ]
            }
          >
            Reply to
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="linkButton-/profile/testuri/post/"
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#f8f3f3",
                    "borderRadius": 6,
                    "height": 8,
                    "overflow": "hidden",
                    "width": 80,
                  },
                  Object {
                    "left": 2,
                    "position": "relative",
                    "top": 1,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#f8f3f3",
                    "height": 8,
                    "width": 80,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 2,
                "flexDirection": "row",
                "marginBottom": 6,
                "marginTop": 2,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#f8f3f3",
              },
            ]
          }
          testID="mutedPostView"
        >
          <
            icon={
              Array [
                "far",
                "eye-slash",
              ]
            }
            style={
              Object {
                "marginRight": 2,
              }
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                },
                undefined,
              ]
            }
          >
            This post is by a muted account.
          </Text>
        </View>
        <View />
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 2,
                  "left": 2,
                  "right": 2,
                  "top": 2,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="postCtrlsReplyButton"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={15}
                bbWidth={15}
                color={4286210401}
                fill="none"
                focusable={false}
                height={15}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke="#7A6161"
                strokeWidth={3}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "color": "#7A6161",
                    },
                    Object {
                      "flex": 0,
                      "height": 15,
                      "width": 15,
                    },
                  ]
                }
                tintColor={4286210401}
                vbHeight={24}
                vbWidth={24}
                width={15}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  stroke={4286210401}
                  strokeWidth={3}
                >
                  <RNSVGPath
                    d="M2.25 12.76c0 1.6 1.123 2.994 2.707 3.227 1.068.157 2.148.279 3.238.364.466.037.893.281 1.153.671L12 21l2.652-3.978c.26-.39.687-.634 1.153-.67 1.09-.086 2.17-.208 3.238-.365 1.584-.233 2.707-1.626 2.707-3.228V6.741c0-1.602-1.123-2.995-2.707-3.228A48.394 48.394 0 0012 3c-2.392 0-4.744.175-7.043.513C3.373 3.746 2.25 5.14 2.25 6.741v6.018z"
                    propList={
                      Array [
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontSize": 17,
                      "letterSpacing": 0.2,
                    },
                    Array [
                      Object {
                        "color": "#7A6161",
                      },
                      Object {
                        "marginLeft": 5,
                      },
                      Object {
                        "fontSize": 15,
                      },
                    ],
                  ]
                }
              >
                0
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 2,
                  "left": 2,
                  "right": 2,
                  "top": 2,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="postCtrlsToggleRepostButton"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <
                  icon="retweet"
                  size={19}
                  style={
                    Object {
                      "color": "#7A6161",
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontSize": 17,
                      "letterSpacing": 0.2,
                    },
                    Array [
                      Object {
                        "color": "#7A6161",
                      },
                      Object {
                        "fontSize": 15,
                      },
                      Object {
                        "marginLeft": 5,
                      },
                    ],
                  ]
                }
              >
                0
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 2,
                  "left": 2,
                  "right": 2,
                  "top": 2,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
              testID="postCtrlsToggleUpvoteButton"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={19}
                  bbWidth={19}
                  color={4286210401}
                  focusable={false}
                  height={19}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "color": "#7A6161",
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": 19,
                        "width": 19,
                      },
                    ]
                  }
                  tintColor={4286210401}
                  vbHeight={14}
                  vbWidth={14}
                  width={19}
                >
                  <RNSVGGroup>
                    <RNSVGPath
                      d="M 7 3 L 2 8 L 4.5 8 L 4.5 11.5 L 9.5 11.5 L 9.5 8 L 12 8 L 7 3 Z"
                      propList={
                        Array [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        Array [
                          2,
                        ]
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth={1.5}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontSize": 17,
                      "letterSpacing": 0.2,
                    },
                    Array [
                      Object {
                        "color": "#7A6161",
                      },
                      Object {
                        "fontSize": 15,
                      },
                      Object {
                        "marginLeft": 5,
                      },
                    ],
                  ]
                }
              >
                0
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
            >
              <
                icon="ellipsis-h"
                size={18}
                style={
                  Array [
                    Object {
                      "marginTop": 2,
                    },
                    Object {
                      "marginRight": 5,
                    },
                    Object {
                      "color": "#c1b9b9",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
